test/common/main_interface_test.cc

```diff
@@ -218,6 +218,75 @@ TEST(MainInterfaceTest, ResetStream) {
+  EXPECT_EQ(ENVOY_FAILURE, send_data(0, c_data, false));
+  EXPECT_EQ(ENVOY_FAILURE, send_trailers(0, c_trailers));
+  EXPECT_EQ(ENVOY_FAILURE, reset_stream(0));
+
+  // Release memory
+  release_envoy_headers(c_headers);
+  c_data.release(c_data.context);
+  release_envoy_headers(c_trailers);
+}
+
+TEST(MainInterfaceTest, RegisterPlatformApi) {
+  // This config is the minimal envoy mobile config that allows for running the engine.
+  const std::string config =
+      "{\"admin\":{},\"static_resources\":{\"listeners\":[{\"name\":\"base_api_listener\","
+      "\"address\":{\"socket_address\":{\"protocol\":\"TCP\",\"address\":\"0.0.0.0\",\"port_"
+      "value\":10000}},\"api_listener\":{\"api_listener\":{\"@type\":\"type.googleapis.com/"
+      "envoy.extensions.filters.network.http_connection_manager.v3.HttpConnectionManager\",\"stat_"
+      "prefix\":\"hcm\",\"route_config\":{\"name\":\"api_router\",\"virtual_hosts\":[{\"name\":"
+      "\"api\",\"include_attempt_count_in_response\":true,\"domains\":[\"*\"],\"routes\":[{"
+      "\"match\":{\"prefix\":\"/"
+      "\"},\"route\":{\"cluster_header\":\"x-envoy-mobile-cluster\",\"retry_policy\":{\"retry_back_"
+      "off\":{\"base_interval\":\"0.25s\",\"max_interval\":\"60s\"}}}}]}]},\"http_filters\":[{"
+      "\"name\":\"envoy.router\",\"typed_config\":{\"@type\":\"type.googleapis.com/"
+      "envoy.extensions.filters.http.router.v3.Router\"}}]}}}]},\"layered_runtime\":{\"layers\":[{"
+      "\"name\":\"static_layer_0\",\"static_layer\":{\"overload\":{\"global_downstream_max_"
+      "connections\":50000}}}]}}";
+  const std::string level = "debug";
+  engine_test_context engine_cbs_context{};
+  envoy_engine_callbacks engine_cbs{[](void* context) -> void {
+                                      auto* engine_running =
+                                          static_cast<engine_test_context*>(context);
+                                      engine_running->on_engine_running.Notify();
+                                    } /*on_engine_running*/,
+                                    [](void* context) -> void {
+                                      auto* exit = static_cast<engine_test_context*>(context);
+                                      exit->on_exit.Notify();
+                                    } /*on_exit*/,
+                                    &engine_cbs_context /*context*/};
+  run_engine(0, engine_cbs, config.c_str(), level.c_str());
+
+  ASSERT_TRUE(
+      engine_cbs_context.on_engine_running.WaitForNotificationWithTimeout(absl::Seconds(10)));
+
+  EXPECT_EQ(ENVOY_SUCCESS, register_platform_api("api", nullptr));
+
+  terminate_engine(0);
+
+  ASSERT_TRUE(engine_cbs_context.on_exit.WaitForNotificationWithTimeout(absl::Seconds(10)));
+}
+
+TEST(MainInterfaceTest, InitEngineReturns1) {
+  // TODO(goaway): return new handle once multiple engine support is in place.
+  // https://github.com/lyft/envoy-mobile/issues/332
+  ASSERT_EQ(1, init_engine());
+}
```

Comment:
This doesn't really feel like a useful test to me. I get that we're just trying to increase coverage, but I don't know that we actually want to be asserting or implying that this returns a specific value. If we were to change this to return 2 tomorrow, everything should still work, and the only thing that would be different is that now we'd have a broken test.

Reply:
I see this as a sentinel that we have to have coverage for `init_engine()`. So in my mind having a test that breaks is, in and of itself, useful.